modules/geo-layers/src/imagery-layer/imagery-layer.ts

```diff
@@ -0,0 +1,208 @@
+// deck.gl, MIT license
+// Attributions:
+// Copyright 2022 Foursquare Labs, Inc.
+
+/* eslint-disable camelcase */ // Some WMS parameters are not in camel case
+/* global setTimeout, clearTimeout */
+
+import {
+  Layer,
+  CompositeLayer,
+  CompositeLayerProps,
+  UpdateParameters,
+  DefaultProps
+} from '@deck.gl/core';
+import {BitmapLayer} from '@deck.gl/layers';
+import type {ImageSourceMetadata, ImageType, ImageServiceType} from '@loaders.gl/wms';
+import {ImageSource, createImageSource} from '@loaders.gl/wms';
+
+export type ImageryLayerProps = CompositeLayerProps<string | ImageSource> & {
+  serviceType?: ImageServiceType | 'auto';
+  layers: string[];
+  onMetadataLoadStart: () => void;
+  onMetadataLoadComplete: (metadata: ImageSourceMetadata) => void;
+  onMetadataLoadError: (error: Error) => void;
+  onImageLoadStart: (requestId: unknown) => void;
+  onImageLoadComplete: (requestId: unknown) => void;
+  onImageLoadError: (requestId: unknown, error: Error) => void;
+};
+
+const defaultProps: ImageryLayerProps = {
+  id: 'imagery-layer',
+  data: undefined!,
+  serviceType: 'auto',
+  layers: undefined!,
+  onMetadataLoadStart: () => {},
+  onMetadataLoadComplete: () => {},
+  // eslint-disable-next-line
+  onMetadataLoadError: (error: Error) => console.error(error),
+  onImageLoadStart: () => {},
+  onImageLoadComplete: () => {},
+  // eslint-disable-next-line
+  onImageLoadError: (requestId: unknown, error: Error) => console.error(error, requestId)
```

Comment:
Define [prop types](https://deck.gl/docs/developer-guide/custom-layers/prop-types#built-in-types) to prevent unnecessary updates in React